### PR TITLE
Adding new package: py-pathos (and dependencies)

### PR DESCRIPTION
**Pull request by JavierCVilla**
Parallel graph management and execution in heterogeneous computing.

This PR includes:
- py-pox
- py-multiprocess
- py-ppft
- py-dill

```
o  py-pathos
|\
| |\
| | |\
| | | |\
| | | | |\
| | o | | |  py-ppft
| |/| | | | 
|/|/| | | | 
| | |\ \ \ \
| | | | |_|/
| | | |/| | 
| | | | o |  py-pox
| |_|_|/| | 
|/| |_|/ /
| |/| | | 
| | | | o  py-multiprocess
| |_|_|/| 
|/| |_|/| 
| |/| |/
| | | o  py-dill
| |_|/| 
|/| |/
| |/| 
| o |  py-setuptools
...
```

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-multiprocess/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    """Better multiprocessing and multithreading in Python"""
+
+    homepage = "https://github.com/uqfoundation/multiprocess"
+    url = "https://pypi.io/packages/source/m/multiprocess/multiprocess-0.70.5.zip"
+
+    version('0.70.5', 'bfe394368b1d98192f1f62cc0060be20')
+    version('0.70.4', '443336d84c574106da6c67d4574b7614')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-dill', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-multiprocess/package.py]:
```python
'py-dill@0.2.6:'
```

File: var/spack/repos/builtin/packages/py-multiprocess/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class PyMultiprocess(PythonPackage):
+    """Better multiprocessing and multithreading in Python"""
+
+    homepage = "https://github.com/uqfoundation/multiprocess"
+    url = "https://pypi.io/packages/source/m/multiprocess/multiprocess-0.70.5.zip"
+
+    version('0.70.5', 'bfe394368b1d98192f1f62cc0060be20')
+    version('0.70.4', '443336d84c574106da6c67d4574b7614')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-multiprocess/package.py]:
```python
depends_on('python@2.6:2.8,3.1:')
```

File: var/spack/repos/builtin/packages/py-multiprocess/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-multiprocess/package.py]:
```python
'py-setuptools@0.6:'
```

File: var/spack/repos/builtin/packages/py-pathos/package.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class PyPathos(PythonPackage):
+    """Parallel graph management and execution in heterogeneous computing """
+
+    homepage = "https://github.com/uqfoundation/pathos"
+    url      = "https://pypi.io/packages/source/p/pathos/pathos-0.2.0.zip"
+
+    version('0.2.0', '7a840ce6c3a67d71e6ad7339034ec53e')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathos/package.py]:
```python
depends_on('python@2.6:2.8,3.1:')
```

File: var/spack/repos/builtin/packages/py-pathos/package.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    depends_on('py-setuptools@0.6:', type='build')
+    depends_on('py-multiprocess@0.70.5:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathos/package.py]:
```python
'@0.70.4:'
```

File: var/spack/repos/builtin/packages/py-pathos/package.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    depends_on('py-pox@0.2.3:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathos/package.py]:
```python
'@0.2.2:'
```

File: var/spack/repos/builtin/packages/py-pathos/package.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    depends_on('py-ppft@1.6.4.7.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathos/package.py]:
```python
'@1.6.4.5:'
```

File: var/spack/repos/builtin/packages/py-pathos/package.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    depends_on('py-dill@0.2.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathos/package.py]:
```python
'@0.2.5:'
```

File: var/spack/repos/builtin/packages/py-pox/package.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+class PyPox(PythonPackage):
+    """Utilities for filesystem exploration and automated builds."""
+
+    homepage = "https://github.com/uqfoundation/pox"
+    url      = "https://pypi.io/packages/source/p/pox/pox-0.2.3.zip"
+
+    version('0.2.3', 'fcdfd9a9ab0f72367258b675554f6a83')
+    version('0.2.2', 'e1e2ce99a63d7226ea3c1a2ce389610d')
+    version('0.2.1', '517dc13c2bc2429d36a0c636f3ce42db')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pox/package.py]:
```python
depends_on('python@2.5:2.8,3.1:')
```

File: var/spack/repos/builtin/packages/py-pox/package.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pox/package.py]:
```python
'py-setuptools@0.6:'
```

File: var/spack/repos/builtin/packages/py-ppft/package.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+class PyPpft(PythonPackage):
+    """Distributed and parallel python """
+
+    homepage = "https://github.com/uqfoundation/ppft"
+    url      = "https://pypi.org/packages/source/p/ppft/ppft-1.6.4.7.1.zip"
+
+    version('1.6.4.7.1',  '2b196a03bfbc102773f849c6b21e617b')
+    version('1.6.4.6',   'e533432bfba4b5a523a07d58011df209')
+    version('1.6.4.5',   'd2b1f9f07eae22b31bfe90f544dd3044')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ppft/package.py]:
```python
depends_on('python@2.5:2.8,3.1:')
```

File: var/spack/repos/builtin/packages/py-pathos/package.py (hunk 5 above)
Comment (JavierCVilla) [inline var/spack/repos/builtin/packages/py-pathos/package.py]: @adamjstewart What are the guidelines that you use to determine the versions?

Most part of versions I put come from the [setup.py](https://github.com/uqfoundation/pathos/blob/master/setup.py#L211), except for some that escaped to me, as in `py-multiprocess`

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathos/package.py]: The guidelines I use are the version requirements from the latest version in Spack. It looks like you used the version requirements for the latest version in GitHub (develop). I'm not opposed to this, although I do think it makes more sense to use the version requirements from the latest version in Spack.

**Pull request merged**